proto/sf/substreams/rpc/v2/service.proto

```diff
@@ -130,6 +130,37 @@ message OutputDebugInfo {
 
 message ModulesProgress {
   repeated ModuleProgress modules = 1;
+  repeated Job running_jobs = 2;
+  repeated ModuleStats modules_stats = 3;
+}
+
+message Job {
+    uint64 stage = 1;
+    repeated string modules = 2;
+    uint64 start_block = 3;
+    uint64 stop_block = 4;
+    uint64 processed_blocks = 5;
+    uint64 duration_secs = 6;
+}
+
+message ModuleStats {
+    string name = 1;
+
+    // total_processed_blocks includes duplicate for each stage going over that module
+    uint64 total_processed_blocks = 2;
+    uint64 highest_processed_block = 3;
+    uint64 total_processing_time_secs = 4;
+    uint64 total_external_calls_time_secs = 5;
+    uint64 total_external_calls_count = 6;
+    uint64 total_store_operations_time_secs = 7;
+    uint64 total_store_reads_count = 8;
+    uint64 total_errors_count = 9;
+
+    // store-specific (will be 0 on mappers)
+    uint64 total_store_writes_count = 10;
+    uint64 total_store_deleteprefix_count = 11;
+    uint64 store_size_bytes = 12;
+    uint64 highest_merged_block = 13;
```

Comment:
Not sure I see what is this metric

Reply:
I'll add a comment: running a store in chunks:
[0-1000] [1000-2000] [missing chunk...] [3000-4000]
the highest-merged-block for that module is 2000. That information is very relevant if you are trying to figure out why you're not receiving any data from a mapper that depends on this module (imagine that block 2012 is excruciatingly slow...)

Reply:
Yes that's good. I'd also want to know the time spent in merging, and/or a way to understand we're blocked on merging, and that's what's slowing things down.

Actually, I'd like to know in the ModuleStats whether the module is currently being merged.. so across the updates, you could see if either the blocks are ramping up, or if the "merging: bool" field is often true.

That information is available on a per-stage basis in the Scheduler.

I'm wondering if we shouldn't have a StageStats rather than a Module Stats.. or perhaps those values nested (`StageStats{modules ModuleStats{{}, {}, {}}`), so you'd look down the Module only if you really need to.

 I imagine the `GUI` to display only stages at some point.. and perhaps as you hover, or select the row, you can see which modules are included in that stage, and get a breakdown (for instance, to know if a given module is still merging, and other modules have stopped, and at the stage level, you would be "merging" all the while any of the module below is merging.

Reply:
We don't need "stats" for stages as much as we need to know the completed blockranges (the jobs are per-stage so they complete the information for progress...)
I'll add `stages` to the modules progress that gives you all the completed ranges.
I'll remove the list of modules from the job, because they can be looked up on the stages.
Stats must be per-module because this is how you figure out issues, but the progress must be per stage.